### PR TITLE
test: fill critical test coverage gaps from code review

### PR DESCRIPTION
**Pull request by Djelibeybi**
## Summary

- Add multizone state update tests verifying `get_color_zones()` and `get_extended_color_zones()` correctly update `_state.zones` (T-C2)
- Add 13 adversarial mDNS DNS parser tests covering empty/truncated packets, inflated record counts, oversized rdlength, compression pointer chains at/beyond max depth, circular pointers, malformed UTF-8 labels, and forward pointers (T-H2, SEC-M2, SEC-L3)
- Add 3 discovery source ID and broadcast serial rejection tests using crafted LIFX packets (T-H3)
- Add `pytest_benchmark_update_machine_info` hook to redact hostname from saved benchmark JSON files

## Test plan

- [x] `uv run --frozen pytest` — 2465 tests pass
- [x] `uv run pyright` — 0 errors, 0 warnings
- [x] `uv run ruff format . && uv run ruff check .` — clean
- [x] Benchmarks saved with redacted hostname

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test_network/test_discovery_errors.py [hunk 1]
```diff
@@ -126,6 +128,159 @@ async def test_discovery_overall_timeout_branch(self) -> None:
...
+
+            raise LifxTimeoutError("timeout")
+
+        with patch("lifx.network.discovery.UdpTransport") as mock_transport_cls:
+            mock_transport = AsyncMock()
+            mock_transport.__aenter__ = AsyncMock(return_value=mock_transport)
+            mock_transport.__aexit__ = AsyncMock(return_value=False)
+            mock_transport.send = AsyncMock()
+            mock_transport.receive = mock_receive
+            mock_transport_cls.return_value = mock_transport
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply tests/test_network/test_discovery_errors.py]: Fixed in `85e5e14` — patched `allocate_source` to a known value (42) and use a different value (999) for the crafted packet, guaranteeing the mismatch deterministically.

File: tests/test_network/test_discovery_errors.py [hunk 2]
```diff
@@ -126,6 +128,159 @@ async def test_discovery_overall_timeout_branch(self) -> None:
...
+
+        # The crafted packet should have been rejected (source mismatch)
+        assert len(devices) == 0
+
+    @pytest.mark.asyncio
+    async def test_broadcast_serial_all_ff_rejected(self) -> None:
+        """Response with all-0xFF broadcast serial must not be yielded.
+
+        Discovery validates that the serial is not the all-0xFF broadcast
+        address used for discovery requests.
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply tests/test_network/test_discovery_errors.py]: Fixed in `85e5e14` — corrected the docstring. Discovery sends with all-zero target but rejects all-0xFF in responses as an invalid broadcast serial.

File: tests/benchmarks/conftest.py [hunk 3]
```diff
@@ -1,3 +1,13 @@
 """Shared fixtures for performance benchmark tests."""
 
 from __future__ import annotations
+
+from typing import Any
+
+
+def pytest_benchmark_update_machine_info(
+    config: Any,
+    machine_info: dict[str, Any],  # noqa: ARG001
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply tests/benchmarks/conftest.py]: Fixed in `21ad6de` — moved `# noqa: ARG001` to `config` (the actually unused parameter).
